### ISSUE TITLE
実装: non-playing AI の反復PPO学習を1コマンドで開始できるorchestrator

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#223]
Parent: #202
Investigation: #201
Depends on: #221
Base branch: `feature/issue-202-non-playing-ai`

## 背景

Issue #221 で、bidding / adjutant / exchange について

rollout生成 -> PPO学習 -> ONNX export -> full-policy評価

を1回だけ縦通しできる `napoleon-run-non-playing-rl` が実装された。

smokeでは、各phaseのdataset生成・学習・ONNX export・runtime利用・full-policy評価まで正常に完走し、fallback=0 / illegal action=0 を確認できた。

ただし現状は1回の更新で終了するため、実際の強化学習として必要な

policy_v0 -> rollout -> PPO update -> policy_v1 -> rollout -> PPO update -> ...

という反復学習loopがない。

## 目的

non-playing AI（bidding / adjutant / exchange）の反復PPO学習を、ユーザーが1コマンドで開始できるようにする。

初回起動から各iterationのrollout / train / export / evaluation / checkpoint管理まで自動で進み、途中停止後もresumeできる構成とする。

## 学習対象

- bidding policy
- adjutant policy
- exchange policy (`sequential-card-v1`)

playing policyは初版では repo管理済み `ppo-separated-v1000` をfrozen固定とする。

## 反復loop

各iterationで原則として以下を行う。

1. 現在のbidding / adjutant / exchange Actor artifactをONNX runtime用に準備
2. current policyでnon-playing rollout datasetを生成
3. 終局role rewardを各phase sampleへ付与
4. bidding PPO update
5. adjutant PPO update
6. exchange PPO update
7. 各phase checkpoint保存
8. 各phase ONNX export
9. iteration summary保存
10. 設定された間隔でfull-policy evaluation
11. 次iterationでは更新済みpolicyをbehavior policyとして使用

## 初期policy

初回iterationでは、#221 のbootstrap方式を再利用してよい。

可能なら既存supervised Actor checkpointを初期Actorとして使えるoptionも残す。ただし初版の必須条件ではない。

Criticは各phase独立とし、前iterationのCriticもcheckpointから継続する。

## CLI

ユーザーが1コマンドで学習を開始できること。

既存 `napoleon-run-non-playing-rl` を拡張するか、長時間学習用に別CLIを追加するかは既存設計との整合性を優先する。

想定例:

```bash
cd python
uv run napoleon-run-non-playing-rl \
  --output-dir ~/napoleon_runs/non-playing-ppo-v1 \
  --iterations 1000 \
  --games-per-iteration 200 \
  --epochs 4 \
  --batch-size 256 \
  --learning-rate 1e-4 \
  --evaluation-every 10 \
  --evaluation-games 100 \
  --seed 202
```

実際の引数名は既存CLIとの整合を優先する。

### resume

同じrun directoryから再開可能にする。

例:

```bash
uv run napoleon-run-non-playing-rl \
  --output-dir ~/napoleon_runs/non-playing-ppo-v1 \
  --resume
```

resume時は、最後に正常完了したiterationとcheckpointを検出し、その次から開始する。

中途半端なiteration directoryがある場合は、安全に未完了として扱い、既存の完了済みartifactを壊さない。

### run directory

例:

```text
<run-dir>/
  config.json
  run-summary.json
  iterations/
    iter-000000/
      bidding/
        dataset/
        input-checkpoint.pt
        output-checkpoint.pt
        policy.onnx
        policy.json
      adjutant/
        ...
      exchange/
        ...
      evaluation.json
      iteration.json
    iter-000001/
      ...
  latest/
    bidding -> ...
    adjutant -> ...
    exchange -> ...
```

実際のlayoutは既存playing PPO orchestratorの方式に合わせてよい。

## rollout

- iterationごとに新しいseed範囲を使う
- 同一run / same seed設定で再現可能なseed progressionにする
- bidding / adjutant / exchangeは各phaseごとにcurrent policyをbehavior policyとして使う
- playingはfrozen `ppo-separated-v1000`
- behavior ONNX / metadata SHAをmanifestへ記録する既存仕様を維持する

## PPO

- #207 / #209 / #210 の既存trainerを再利用する
- PPOアルゴリズムを重複実装しない
- previous output checkpointを次iterationのinput checkpointとして継続する
- ActorだけでなくCriticも継続学習する
- gamma / reward version / PPO clip等の設定をrun configへ保存する

## evaluation

定期的にfull-policy evaluationを実施する。

最低限:

- completed / failed games
- fallback count
- illegal action count
- phase別 ONNX call count
- bidding pass率 / bid 13〜19分布 / suit分布
- 平均宣言枚数
- 宣言成功率
- role別平均reward
- adjutant選択分布
- exchange傾向

可能なら RuleBased non-playing / Critic-EV bidding baselineとの比較も残す。

## logging / progress

長時間学習中に進捗を追えるよう、iterationごとに少なくとも以下を表示する。

- iteration number
- rollout games / sample count
- phase別 mean reward
- phase別 PPO loss / value loss / clipped fraction
- evaluation実施時の主要指標
- elapsed time
- latest checkpoint/artifact path

## safety

- 既存run directoryを誤って上書きしない
- 新規runとresumeを明確に区別する
- config mismatch時はfail-closeする
- playing policy artifact / reward version / model schema / exchange decision modeの不一致を検出する
- iteration完了前にcompleted markerを書かない

## 最初の実運用設定

実装後、ユーザーがそのまま実行できる推奨コマンドを提示する。

初回の実験では過度に大規模にせず、例として以下程度を基準にしてよい。

- iterations: 100〜1000
- games per iteration: 200
- epochs: 4
- learning rate: 1e-4
- batch size: 128〜256程度
- evaluation every: 10 iterations
- evaluation games: 100程度

ただし最終値は既存playing PPO設定やsample数を確認して合理的に選ぶ。

## Acceptance Criteria

- 1コマンドで反復non-playing PPO学習を開始できる
- iteration 0 -> 1 -> 2 と更新済みpolicyを使ってrolloutが継続する
- bidding / adjutant / exchangeのActorとCriticが前iteration checkpointから継続される
- iterationごとのdataset / checkpoint / ONNX / metadata / summaryが保存される
- current behavior artifactのSHA/provenanceがdatasetに記録される
- 定期full-policy evaluationが自動実行される
- fallback count / illegal action countが記録される
- run configとseed progressionが保存される
- 中断後に `--resume` 等で最後の完了iterationから安全に再開できる
- config mismatchやartifact mismatchはfail-closeする
- 2〜3 iteration程度のend-to-end smoke testが通る
- 既存1-shot smoke CLIの用途を壊さない、または互換な使い方を明示する
- READMEに「新規学習開始」「resume」「出力確認」の具体例を追加する
- 実装完了時に、ユーザーがそのままコピペできる本番学習開始コマンドを提示する

## 非目標

- playing policyのjoint training
- bidding / adjutant / exchangeのshared network化
- reward値の再設計
- PPOアルゴリズム自体の変更
- opponent poolの高度化
- C++ backendへの全面移植

## PR運用

このIssueは #202 の追加子Issueとして扱う。
専用ブランチを最新の `feature/issue-202-non-playing-ai` から作り、PRのbaseも `feature/issue-202-non-playing-ai` とする。

親PR #203 はmainへまだMergeしない。
